coriolis/tests/api/middleware/test_auth.py

```diff
@@ -0,0 +1,182 @@
+    def test__call__(
+        self,
+        mock_get_user,
+        mock__get_project_id,
+        mock_request_context
+    ):
+        req_mock = mock.Mock()
+        mock__get_project_id.return_value = 'mock_tenant'
+        mock_get_user.return_value = "mock_user"
+
+        expected_roles = ['1', '2', '3']
+        expected_project_name = 'mock_project_name'
+        expected_project_domain_name = 'mock_project_domain_name'
+        expected_user_domain_name = 'mock_user_domain_name'
+        expected_auth_token = 'mock_token123'
+        expected_remote_address = 'mock_addr'
+        expected_service_catalog = {"catalog1": ["service1", "service2"]}
+        expected_req_id = 'mock_req_id'
+
+        req_mock.remote_addr = expected_remote_address
+        req_mock.environ = {
+            request_id.ENV_REQUEST_ID: expected_req_id.encode()
+        }
+
+        req_mock.headers = {
+            'X_ROLE': '1,2,3',
+            'X_TENANT_NAME': expected_project_name,
+            'X-Project-Domain-Name': expected_project_domain_name,
+            'X-User-Domain-Name': expected_user_domain_name,
+            'X_AUTH_TOKEN': expected_auth_token,
+            'X_SERVICE_CATALOG':
+                str(expected_service_catalog).replace("'", '"'),
+        }
+
+        result = self.auth(req_mock)
+
+        self.assertEqual(
+            CoriolisKeystoneContext(wsgi.Middleware).application,
+            result
+        )
+
+        mock_get_user.assert_called_once()
+        mock__get_project_id.assert_called_once()
+        mock_request_context.assert_called_once_with(
+            "mock_user",
+            "mock_tenant",
+            project_name=expected_project_name,
+            project_domain_name=expected_project_domain_name,
+            user_domain_name=expected_user_domain_name,
+            roles=expected_roles,
+            auth_token=expected_auth_token,
+            remote_address=expected_remote_address,
+            service_catalog=expected_service_catalog,
+            request_id=expected_req_id
+        )
+
+        self.assertEqual(
+            req_mock.environ['coriolis.context'],
+            mock_request_context.return_value
+        )
+
+    @mock.patch.object(CoriolisKeystoneContext, "_get_project_id")
+    @mock.patch.object(CoriolisKeystoneContext, "_get_user")
+    def test__call__invalid_service_catalog(
+        self,
+        mock_get_user,
+        mock__get_project_id,
+    ):
+        req_mock = mock.Mock()
+        mock__get_project_id.return_value = 'mock_tenant'
+        mock_get_user.return_value = "mock_user"
+        req_mock.environ.return_value = "mock_environ"
+
+        expected_project_name = 'mock_project_name'
+        expected_project_domain_name = 'mock_project_domain_name'
+        expected_user_domain_name = 'mock_user_domain_name'
+        expected_auth_token = 'mock_token123'
+        expected_service_catalog = {"catalog1": ["service1", "service2"]}
```

Comment:
You don't need to add values to mock attributes just to pass an invalid service_catalog. Only keep the service_catalog values, it shouldn't raise any errors (you mock `_get_project_id` and `_get_user` methods)